lexibank_kitchensemitic.py

```diff
@@ -1,187 +1,110 @@
-                    # it looks like the cognate set used is the first one, so fill the
-                    # cognates to the same size with the token used as a
-                    # non-cognate marker ('-')
-                    if len(forms) > len(cogs):
-                        cogs.extend('-' for _ in range(len(forms) - len(cogs)))
-                    
-                    # there does appear to be one instance of more cognates than forms:
-                    #   Harsusi's "ges'" - RAIN (N.) = D/I
-                    # .. but it looks cognate with Hebrew's and Ugaritic's set D rather than
-                    # set I (e.g. Ogaden Arabic "matar"), so I think it's a typo and
-                    # we'll ignore it.
-                    
-                    # there are also forms that have no lexeme but do have a cognate set...
-                    # have to ignore these for now. Can't think of what to do with them
-                    # otherwise:
-                    #    Mɛhri Dog --- D
-                    #    Mɛhri Dry (adj.) --- F
-                    #    Hebrew Grass --- C/J
-                    #    Hebrew New --- A
-                    #    Mɛhri Seed --- C
-                    #    Hebrew Three --- A
-                    #    Mɛhri Warm --- H
-                    #    Hebrew Ye --- A
-
-                    for form, cog in zip(forms, cogs):
-                        assert len(form), "Empty form %s - %r - %r - %r" % (lid, gloss, form, cog)
-                        assert len(cog), "Empty cognate %s - %r - %r - %r" % (lid, gloss, form, cog)
-                        
-                        cogid = '%s-%s' % (csid, cog)
-                        for row in ds.add_lexemes(
-                                Language_ID=lid,
-                                Parameter_ID=csid,
-                                Value=form,
-                                Source=src
-                            ):
-                            if cog != '-':
-                                ds.add_cognate(lexeme=row, Cognateset_ID=cogid, Source=['Kitchen2009'])
+    language_class = KSLanguage
+
+    form_spec = FormSpec(
+        brackets={}, separators="/,", missing_data=("---",), strip_inside_brackets=False
+    )
+
+    def cmd_download(self, args):
+        self.raw_dir.xls2csv("Semitic.Wordlists.xls")
+        self.raw_dir.xls2csv("Semitic.Codings.Multistate.xlsx")
+
+    def cmd_makecldf(self, args):
+        lexeme_rows = self.raw_dir.read_csv("Semitic.Wordlists.ActualWordlists.csv")
+        lexeme_header = lexeme_rows.pop(0)
+
+        cognate_table = make_cognate_table(
+            self.raw_dir.read_csv("Semitic.Codings.Multistate.Sheet1.csv")
+        )
+
+        languages, concepts = {}, {}
+
+        for concept in self.conceptlist.concepts.values():
+            concepts[concept.english] = concept.id
+
+        args.writer.add_concepts(id_factory=lambda c: c.id)
+
+        for language in self.languages:
```

Comment:
Ah, alright, so if languages are anyway looped across, one could as well use `args.writer.add_language()` explicitly here. I have no real preference. What do the others think?

Reply:
Happy with either. Looks like we're making a name here that isn't in languages.csv

Reply:
Happy with either as well, but `writer.add_language()` feels a bit more consistent with the library.